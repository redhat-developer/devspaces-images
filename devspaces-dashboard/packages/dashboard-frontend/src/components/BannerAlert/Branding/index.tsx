/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { Banner } from '@patternfly/react-core';
import React from 'react';
import { connect, ConnectedProps } from 'react-redux';
import sanitizeHtml from 'sanitize-html';
import { AppState } from '../../../store';
import { selectBranding } from '../../../store/Branding/selectors';

type Props = MappedProps;

class BannerAlertBranding extends React.PureComponent<Props> {
  render() {
    const warningMessage = this.props.branding.header?.warning;
    if (!warningMessage) {
      return null;
    }

    const warningMessageHTML = sanitizeHtml(warningMessage, {
      allowedTags: ['a'],
      allowedAttributes: {
        a: ['href', 'target'],
      },
      allowedSchemes: ['http', 'https'],
    });

    return (
      <Banner className="pf-u-text-align-center" variant="warning">
        <div dangerouslySetInnerHTML={{ __html: warningMessageHTML }}></div>
      </Banner>
    );
  }
}

const mapStateToProps = (state: AppState) => ({
  branding: selectBranding(state),
});

const connector = connect(mapStateToProps);

type MappedProps = ConnectedProps<typeof connector>;
export default connector(BannerAlertBranding);
