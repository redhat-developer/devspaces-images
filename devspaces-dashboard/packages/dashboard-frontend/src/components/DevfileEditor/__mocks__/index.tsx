/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React, { ChangeEvent, ClipboardEvent } from 'react';
import { safeDump } from 'js-yaml';
import devfileApi from '../../../services/devfileApi';
import stringify from '../../../services/helpers/editor';

type Props = {
  devfile: che.WorkspaceDevfile | devfileApi.Devfile;
  decorationPattern?: string;
  onChange: (newValue: string, isValid: boolean) => void;
  isReadonly?: boolean;
};
type State = {
  errorMessage: string;
};

export default class DevfileEditor extends React.PureComponent<Props, State> {
  private editor: HTMLTextAreaElement;

  constructor(props: Props) {
    super(props);

    this.state = {
      errorMessage: '',
    };
  }

  public updateContent(devfile: che.WorkspaceDevfile | devfileApi.Devfile): void {
    if (!this.editor) {
      return;
    }
    this.editor.value = stringify(devfile);
  }

  public componentDidMount(): void {
    const element = document.querySelector('#devfile-editor');
    if (element) {
      const value = stringify(this.props.devfile);
      this.editor = element as HTMLTextAreaElement;
      this.editor.value = value;
    }
  }

  public render(): React.ReactElement {
    const { devfile } = this.props;
    const content = safeDump(devfile);

    const onChange = (event: ChangeEvent<HTMLTextAreaElement>) => {
      const newContent = event.currentTarget.value;
      this.onChange(newContent, true);
    };

    const onPaste = (event: ClipboardEvent<HTMLTextAreaElement>) => {
      const newContent = event.clipboardData?.getData('Text') || '';
      this.onChange(newContent, true);
    };

    return (
      <div className="devfile-editor">
        <div className="monaco">
          <textarea
            id="devfile-editor"
            onChange={onChange}
            onPaste={onPaste}
            value={content}
          ></textarea>
        </div>
        <div className="error"></div>
        <a target="_blank" rel="noopener noreferrer">
          Devfile Documentation
        </a>
      </div>
    );
  }

  private onChange(newValue: string, isValid: boolean): void {
    this.props.onChange(newValue, isValid);
  }
}
