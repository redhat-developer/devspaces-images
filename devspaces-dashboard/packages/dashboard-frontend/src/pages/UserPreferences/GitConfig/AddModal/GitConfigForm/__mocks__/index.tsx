/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/UserPreferences/GitConfig/AddModal/GitConfigForm';
import * as GitConfigStore from '@/store/GitConfig';

export class GitConfigForm extends React.PureComponent<Props> {
  render() {
    const { onChange } = this.props;

    return (
      <div>
        <input
          data-testid="submit-invalid-git-config"
          type="button"
          onClick={() =>
            onChange({ user: { name: 'User One' } } as GitConfigStore.GitConfig, false)
          }
        />
        <input
          data-testid="submit-valid-git-config"
          type="button"
          onClick={() =>
            onChange({ user: { email: 'user-1@chetest.com', name: 'User One' } }, true)
          }
        />
      </div>
    );
  }
}
