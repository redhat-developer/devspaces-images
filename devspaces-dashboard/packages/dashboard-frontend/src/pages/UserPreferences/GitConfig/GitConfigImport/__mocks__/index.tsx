/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/UserPreferences/GitConfig/GitConfigImport';

export class GitConfigImport extends React.PureComponent<Props> {
  render() {
    const { content, onChange } = this.props;

    return (
      <div>
        <input
          data-testid="submit-invalid-git-config"
          type="button"
          value={content}
          onClick={() => onChange('[user]\n\tname = User One\n', false)}
        />
        <input
          data-testid="submit-valid-git-config"
          type="button"
          onClick={() =>
            onChange('[user]\n\tname = User One\n\temail = user-1@chetest.com\n', true)
          }
        />
      </div>
    );
  }
}
