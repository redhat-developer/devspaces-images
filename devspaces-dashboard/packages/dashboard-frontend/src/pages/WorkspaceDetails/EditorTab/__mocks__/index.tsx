/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import common from '@eclipse-che/common';
import React from 'react';
import { Props, State } from '..';
import devfileApi from '../../../../services/devfileApi';
import { Workspace } from '../../../../services/workspace-adapter';

export default class EditorTab extends React.PureComponent<Props, State> {
  cancelChanges: () => void;

  constructor(props: Props) {
    super(props);

    this.state = {
      devfile: {} as devfileApi.Devfile,
      hasChanges: false,
      isDevfileValid: true,
      hasRequestErrors: false,
      currentRequestError: '',
      isExpanded: false,
      showDevfileV2ConfirmationModal: false,
    };

    this.cancelChanges = () => {
      // no-op
    };
  }

  async handleOnSave(workspace: Workspace) {
    try {
      await this.props.onSave(workspace);
    } catch (e) {
      this.setState({
        currentRequestError: common.helpers.errors.getMessage(e),
      });
    }
  }

  render() {
    return (
      <div>
        Fake Editor Tab
        <button onClick={() => this.handleOnSave(this.props.workspace)}>Save</button>
        <button onClick={() => this.props.onDevWorkspaceWarning()}>Warning</button>
        <span data-testid="current-request-error">{this.state.currentRequestError}</span>
      </div>
    );
  }
}
