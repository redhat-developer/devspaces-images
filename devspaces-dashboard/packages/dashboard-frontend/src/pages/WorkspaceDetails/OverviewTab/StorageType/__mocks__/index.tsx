/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { Props, State } from '..';

export default class StorageTypeFormGroup extends React.PureComponent<Props, State> {
  render() {
    return <div>Fake Storage Type Form</div>;
  }
}
