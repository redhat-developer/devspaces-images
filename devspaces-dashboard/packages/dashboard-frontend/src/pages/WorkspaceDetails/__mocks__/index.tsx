/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { Props, State } from '..';

export const mockShowAlert = jest.fn();
export class WorkspaceDetails extends React.PureComponent<Props, State> {
  render() {
    if (!this.props.workspace) {
      return <div>Not found</div>;
    }
    const workspace = this.props.workspace;
    return (
      <div className="WorkspaceDetails">
        Workspace Details Page
        <div data-testid="props-is-loading">{this.props.isLoading}</div>
        <div data-testid="props-old-workspace-path">
          {this.props.oldWorkspaceLocation?.pathname}
        </div>
        <div data-testid="props-show-convert-button">
          {this.props.showConvertButton === true ? 'true' : 'false'}
        </div>
        <div data-testid="props-workspace-id">{this.props.workspace.id}</div>
        <div data-testid="props-workspace-name">{this.props.workspace.name}</div>
        <div data-testid="props-workspaces-link">{this.props.workspacesLink}</div>
        <button onClick={() => this.props.onSave(workspace)}>Save</button>
        <button onClick={() => this.props.onConvert(workspace)}>Convert</button>
      </div>
    );
  }
}
