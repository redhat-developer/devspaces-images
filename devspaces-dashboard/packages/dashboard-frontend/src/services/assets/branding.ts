/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import axios from 'axios';
import { BrandingData } from '../bootstrap/branding.constant';

export async function fetchBranding(url: string): Promise<BrandingData> {
  try {
    const response = await axios.get<BrandingData>(url);
    return response.data;
  } catch (e) {
    throw new Error(`Failed to fetch branding data by URL: ${url}`);
  }
}
