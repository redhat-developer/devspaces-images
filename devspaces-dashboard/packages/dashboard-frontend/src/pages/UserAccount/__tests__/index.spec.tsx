/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { createHashHistory } from 'history';
import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { Store } from 'redux';
import { UserAccount } from '..';
import { FakeStoreBuilder } from '../../../store/__mocks__/storeBuilder';
import { BrandingData } from '../../../services/bootstrap/branding.constant';
import { selectBranding } from '../../../store/Branding/selectors';
import { selectUserProfile } from '../../../store/UserProfile/selectors';

describe('UserAccount page', () => {
  const history = createHashHistory();

  const getComponent = (store: Store): React.ReactElement => {
    const state = store.getState();
    const branding = selectBranding(state);
    const userProfile = selectUserProfile(state);

    return (
      <Provider store={store}>
        <UserAccount
          history={history}
          branding={branding}
          userProfile={userProfile}
          dispatch={jest.fn()}
        />
      </Provider>
    );
  };

  it('should correctly render the component without profile data', () => {
    const store = new FakeStoreBuilder()
      .withBranding({
        name: 'test',
      } as BrandingData)
      .build();
    const component = getComponent(store);
    render(component);

    const editAccountButton = screen.queryByLabelText('edit account info');
    expect(editAccountButton).toBeTruthy();
    expect(editAccountButton).toBeDisabled();

    const json = renderer.create(component).toJSON();

    expect(json).toMatchSnapshot();
  });

  it('should correctly render the component which contains profile data', () => {
    const store = new FakeStoreBuilder()
      .withBranding({
        name: 'Product name',
      } as BrandingData)
      .withUserProfile({
        attributes: {
          firstName: 'John',
          lastName: 'Doe',
          preferred_username: 'Johnny',
        },
        email: 'johndoe@test.com',
        userId: 'john-doe-id',
      })
      .build();
    const component = getComponent(store);
    render(component);

    const editAccountButton = screen.getByLabelText('edit account info');
    expect(editAccountButton).toBeDisabled();

    const json = renderer.create(component).toJSON();

    expect(json).toMatchSnapshot();
  });
});
