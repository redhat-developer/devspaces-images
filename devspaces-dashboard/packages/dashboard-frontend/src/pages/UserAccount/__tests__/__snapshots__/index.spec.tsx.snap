// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAccount page should correctly render the component which contains profile data 1`] = `
<section
  className="pf-c-page__main-section pf-m-light pf-m-fill"
>
  <div
    className="pf-l-stack pf-m-gutter"
  >
    <div
      className="pf-l-stack__item"
    >
      <h1
        className="pf-c-title pf-m-2xl"
      >
        Account
      </h1>
      <p
        className=""
        data-pf-content={true}
      >
        This is where you can view and edit your account information for Product name.
      </p>
    </div>
    <div
      className="pf-l-stack__item pf-m-fill"
    >
      <form
        className="pf-c-form pf-m-limit-width"
        noValidate={true}
      >
        <div
          className="pf-c-form__group"
        >
          <div
            className="pf-c-form__group-label"
          >
            <label
              className="pf-c-form__label"
              htmlFor="form-group-login"
            >
              <span
                className="pf-c-form__label-text"
              >
                Login
              </span>
            </label>
             
          </div>
          <div
            className="pf-c-form__group-control"
          >
            <input
              aria-invalid={false}
              aria-label="readonly login"
              className="pf-c-form-control"
              disabled={true}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              readOnly={false}
              required={false}
              type="text"
              value="Johnny"
            />
            
          </div>
        </div>
        <div
          className="pf-c-form__group"
        >
          <div
            className="pf-c-form__group-label"
          >
            <label
              className="pf-c-form__label"
              htmlFor="form-group-email"
            >
              <span
                className="pf-c-form__label-text"
              >
                Email
              </span>
            </label>
             
          </div>
          <div
            className="pf-c-form__group-control"
          >
            <input
              aria-invalid={false}
              aria-label="readonly email"
              className="pf-c-form-control"
              disabled={true}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              readOnly={false}
              required={false}
              type="text"
              value="johndoe@test.com"
            />
            
          </div>
        </div>
        <div
          className="pf-c-form__group"
        >
          <div
            className="pf-c-form__group-label"
          >
            <label
              className="pf-c-form__label"
              htmlFor="form-group-first-name"
            >
              <span
                className="pf-c-form__label-text"
              >
                First Name
              </span>
            </label>
             
          </div>
          <div
            className="pf-c-form__group-control"
          >
            <input
              aria-invalid={false}
              aria-label="readonly first name"
              className="pf-c-form-control"
              disabled={true}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              readOnly={false}
              required={false}
              type="text"
              value="John"
            />
            
          </div>
        </div>
        <div
          className="pf-c-form__group"
        >
          <div
            className="pf-c-form__group-label"
          >
            <label
              className="pf-c-form__label"
              htmlFor="form-group-last-name"
            >
              <span
                className="pf-c-form__label-text"
              >
                Last Name
              </span>
            </label>
             
          </div>
          <div
            className="pf-c-form__group-control"
          >
            <input
              aria-invalid={false}
              aria-label="readonly last name"
              className="pf-c-form-control"
              disabled={true}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              readOnly={false}
              required={false}
              type="text"
              value="Doe"
            />
            
          </div>
        </div>
        <div
          className="pf-c-form__group pf-m-action"
        >
          <div
            className="pf-c-form__group-control"
          >
            <div
              className="pf-c-form__actions"
            >
              <button
                aria-disabled={true}
                aria-label="edit account info"
                className="pf-c-button pf-m-primary pf-m-disabled"
                data-ouia-component-id="OUIA-Generated-Button-primary-4"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={true}
                onClick={[Function]}
                role={null}
                tabIndex={null}
                type="button"
              >
                Edit
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</section>
`;

exports[`UserAccount page should correctly render the component without profile data 1`] = `
<section
  className="pf-c-page__main-section pf-m-light pf-m-fill"
>
  <div
    className="pf-l-stack pf-m-gutter"
  >
    <div
      className="pf-l-stack__item"
    >
      <h1
        className="pf-c-title pf-m-2xl"
      >
        Account
      </h1>
      <p
        className=""
        data-pf-content={true}
      >
        This is where you can view and edit your account information for test.
      </p>
    </div>
    <div
      className="pf-l-stack__item pf-m-fill"
    >
      <form
        className="pf-c-form pf-m-limit-width"
        noValidate={true}
      >
        <div
          className="pf-c-form__group"
        >
          <div
            className="pf-c-form__group-label"
          >
            <label
              className="pf-c-form__label"
              htmlFor="form-group-login"
            >
              <span
                className="pf-c-form__label-text"
              >
                Login
              </span>
            </label>
             
          </div>
          <div
            className="pf-c-form__group-control"
          >
            <input
              aria-invalid={false}
              aria-label="readonly login"
              className="pf-c-form-control"
              disabled={true}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              readOnly={false}
              required={false}
              type="text"
              value=""
            />
            
          </div>
        </div>
        <div
          className="pf-c-form__group"
        >
          <div
            className="pf-c-form__group-label"
          >
            <label
              className="pf-c-form__label"
              htmlFor="form-group-email"
            >
              <span
                className="pf-c-form__label-text"
              >
                Email
              </span>
            </label>
             
          </div>
          <div
            className="pf-c-form__group-control"
          >
            <input
              aria-invalid={false}
              aria-label="readonly email"
              className="pf-c-form-control"
              disabled={true}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              readOnly={false}
              required={false}
              type="text"
            />
            
          </div>
        </div>
        <div
          className="pf-c-form__group"
        >
          <div
            className="pf-c-form__group-label"
          >
            <label
              className="pf-c-form__label"
              htmlFor="form-group-first-name"
            >
              <span
                className="pf-c-form__label-text"
              >
                First Name
              </span>
            </label>
             
          </div>
          <div
            className="pf-c-form__group-control"
          >
            <input
              aria-invalid={false}
              aria-label="readonly first name"
              className="pf-c-form-control"
              disabled={true}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              readOnly={false}
              required={false}
              type="text"
              value=""
            />
            
          </div>
        </div>
        <div
          className="pf-c-form__group"
        >
          <div
            className="pf-c-form__group-label"
          >
            <label
              className="pf-c-form__label"
              htmlFor="form-group-last-name"
            >
              <span
                className="pf-c-form__label-text"
              >
                Last Name
              </span>
            </label>
             
          </div>
          <div
            className="pf-c-form__group-control"
          >
            <input
              aria-invalid={false}
              aria-label="readonly last name"
              className="pf-c-form-control"
              disabled={true}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              readOnly={false}
              required={false}
              type="text"
              value=""
            />
            
          </div>
        </div>
        <div
          className="pf-c-form__group pf-m-action"
        >
          <div
            className="pf-c-form__group-control"
          >
            <div
              className="pf-c-form__actions"
            >
              <button
                aria-disabled={true}
                aria-label="edit account info"
                className="pf-c-button pf-m-primary pf-m-disabled"
                data-ouia-component-id="OUIA-Generated-Button-primary-2"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={true}
                onClick={[Function]}
                role={null}
                tabIndex={null}
                type="button"
              >
                Edit
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</section>
`;
